### ISSUE TITLE
IRteParam missing types?

### ISSUE THREAD

**Issue by flippidippi** [contentstack/app-sdk#114]
It seems like `IRteParam` is missing types, like `rte.getFieldConfig` and more.
We are trying to create a custom RTE plugin and need to get the field config but need to use `@ts-expect-error` to access.

**Comment by abhishek305** [comment]
Hi @flippidippi,
We do not have handler support for `rte.getFieldConfig()`, did you mean `rte.getConfig`?
For `rte.getConfig` we do not have type support added. 
This is planned for the next release.
Let me know if you have any further questions.

**Comment by flippidippi** [comment]
This is the example I have. If we needed to be able to control the color of the highlight at a field level config.

```tsx
const Highlight = RTE('highlight', rte => {
    // Get the field config
    // @ts-expect-error
    const fieldConfig = rte.getFieldConfig()

    return {
      title: 'Highlight',
      icon: <Icon style={{ padding: '0 6px' }} icon='Edit' size='original' />,
      display: ['toolbar', 'hoveringToolbar'],
      elementType: ['text'],
      render: (props: any) => {
        return (
          <span style={{ background: fieldConfig?.color }}>
            {props.children}
          </span>
        )
      }
    }
  })
```

**Comment by abhishek305** [comment]
Hi @flippidippi 
Thanks for the snippet, but we do not have any handler support for `rte. getFieldConfig`
Its `rte.getConfig` and we'll have a fix for type missing for `rte.getConfig` handler in next release.
Let me know if there's anything I can help with.

**Comment by flippidippi** [comment]
So does `rte.getConfig` get the config at the field level? Or is there a supported way to do that?

**Comment by abhishek305** [comment]
> So does `rte.getConfig` get the config at the field level? Or is there a supported way to do that?

Yes, @flippidippi `rte.getConfig` retrieves the values set in config UI location.
For Eg --> If you are saving any color i.e. `fieldConfig.color`
Similar to shown in above comments.

**Comment by flippidippi** [comment]
I actually don't see `rte.getConfig` in the types either. And this is a screenshot of the difference between the data I get back from each. I don't see the field config in the `rte.getConfig`, neither seem to be on `IRteParam`.

![image](https://github.com/user-attachments/assets/90cfa612-a2ee-4478-a54b-50bb1a74e425)

**Comment by abhishek305** [comment]
@flippidippi the type support for `rte.getConfig` in `IRteParam` would be a part of next release.
An example usage post-release for reference

```
 const examplePlugin = RTE("testApp", async (rte) => {
       const config = await (rte as IRteParam).getConfig();
}
```

Where IRteParams would have `getConfig` as `getConfig: <T>() => { [key: string]: T };`

Will keep you posted once it is released.

**Comment by abhishek305** [comment]
Closing this issue as it is addressed in app SDK v2.0.3
